ash/src/extensions/ext/descriptor_buffer.rs

```diff
@@ -0,0 +1,224 @@
+use crate::prelude::*;
+use crate::vk;
+use crate::{Device, Instance};
+use std::ffi::CStr;
+use std::mem;
+
+/// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VK_EXT_descriptor_buffer.html>
+#[derive(Clone)]
+pub struct DescriptorBuffer {
+    handle: vk::Device,
+    fp: vk::ExtDescriptorBufferFn,
+}
+
+impl DescriptorBuffer {
+    pub fn new(instance: &Instance, device: &Device) -> Self {
+        let handle = device.handle();
+        let fp = vk::ExtDescriptorBufferFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(handle, name.as_ptr()))
+        });
+        Self { handle, fp }
+    }
+
+    #[inline]
+    pub unsafe fn get_properties(
+        instance: &Instance,
+        pdevice: vk::PhysicalDevice,
+    ) -> vk::PhysicalDeviceDescriptorBufferPropertiesEXT {
+        let mut props_db = vk::PhysicalDeviceDescriptorBufferPropertiesEXT::default();
+        {
+            let mut props = vk::PhysicalDeviceProperties2::default().push_next(&mut props_db);
+            instance.get_physical_device_properties2(pdevice, &mut props);
+        }
+        props_db
+    }
```

Comment:
After checking, most of the extensions _do not_ implement this helper. I feel like we should push users to _not_ call it in favour of retrieving all their needed properties in a single up-front call with all the necessary structures.

Shall I remove them as a breaking change on `main`?

Reply:
I wasn't aware we had any of these, and I agree that a single call makes more sense, if only because it's comparably ergonomic.

Reply:
Removed from this PR, will set up a PR (that obviously won't be backported to 0.37) to remove the rest.